### PR TITLE
parse stav hall mealtime report data

### PR DESCRIPTION
**Pull request by drewvolz**
Parses [Stav Mealtime data](https://www.stolaf.edu/apps/mealtimes/) into a json API

#### Usage

```sh
 ➜ curl http://localhost:3000/v1/reports/stav | jq
 
 # abridged array for demo, expect up to 7 charts
[{
  "title": "Saturday, May 11",
  "times": ["07:45:00","08:00:00","08:15:00","08:30:00","08:45:00","09:00:00","09:15:00","10:45:00","11:00:00","11:15:00","11:30:00","11:45:00","12:00:00","12:15:00","12:30:00","12:45:00","01:00:00","01:15:00","01:30:00","04:15:00","04:30:00","04:45:00","05:00:00","05:15:00","05:30:00","05:45:00","06:00:00","06:15:00","06:30:00","06:45:00","07:00:00","07:15:00","07:30:00"],
  "data": [9,18,15,38,51,41,24,124,142,122,77,98,105,122,112,103,108,78,1,17,44,86,93,110,120,108,128,99,123,84,128,93,1]
  },
  ...
]
 ```

#### Potential applications

- Busyness charts/data in stav menu
- Busyness charts/data in stav hours
- Historical data collection (if we start saving it)

#### Todo
- [x] return the data in a promise because the window dom event isn’t fast enough
- [x] return a correctly typed error message object if an error occurs
- [x] assertion tests

### CODE REVIEW

File: source/reports-bonapp/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * @param {string|URL} reportUrl
+ * @returns {Promise<StavReportType>}
+ */
+export function report(reportUrl) {
+	try {
+		return _report(reportUrl)
+	} catch (err) {
+		console.error(err)
+		Sentry.isInitialized() && Sentry.captureException(err)
+		return CustomCafe({message: 'Could not load BonApp report'})
```
Comment (drewvolz) [inline source/reports-bonapp/index.js:107]: Need a different return type

File: source/reports-bonapp/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			} catch (err) {
+				console.warn({error: err.message})
+			}
+		}
+
+		// TODO: we have data here but we can't return it back fast enough
+		// while in the dom.window.onload callback to the `report` function
+		console.log(StavReportType.parse(payload))
+
+		return StavReportType.parse(payload)
```
Comment (drewvolz) [inline source/reports-bonapp/index.js:93]: This doesn’t return soon enough for our API to give the client the data

File: source/reports-bonapp/index.js (hunk 1 above)
Comment (drewvolz) [reply source/reports-bonapp/index.js]: Updated!

File: source/reports-bonapp/index.js (hunk 2 above)
Comment (drewvolz) [reply source/reports-bonapp/index.js]: Fixed with a promise!

**Review by hawkrives — APPROVED** [review]

**Comment by drewvolz** [comment]
I'm thinking about creating groupings of this data on the server such that the default we return is hourly instead of every 15 minutes

**Review by rye — APPROVED** [review]

**Pull request merged**